solutions/usaco-616.mdx

```diff
@@ -14,42 +14,117 @@ author: Óscar Garries
 
 ```cpp
 #include <bits/stdc++.h>
-
 using namespace std;
 
-typedef long long ll;
-typedef long double ld;
-
-#define f first;
-#define s second;
-
-#define len(x) (int)x.size();
-
-int main(){
+int main() {
 	freopen("cbarn.in", "r", stdin);
 	freopen("cbarn.out", "w", stdout);
-	ios_base::sync_with_stdio(0); cin.tie(0);
+	cin.sync_with_stdio(0); cin.tie(0);
+
 	int n;
 	cin >> n;
-	vector<int> a(n);
-	int cows = 0;
-	for (int i = 0; i < n; i++) {
-		cin >> a[i];
-		cows += a[i];
+
+	vector<int> cows(n);
+
+	for(int i = 0; i < n; i++) {
+		cin >> cows[i];
 	}
-	int sol = INT_MAX;
-	for (int start = 0; start < n; start++) {
-		int cur = 0, cur_cows = cows;
-		for (int i = 0; i < n; i++) {
-			cur_cows -= a[(start + i) % n];
-			cur += cur_cows;
+
+	// run a simple brute force algo, checking each possible barn we can open
+
+	int ret = (int)1e9; // stores the current answer
+	// initially set to an abnormally large number
+
+	for(int open_barn = 0; open_barn < n; open_barn++) {
+		int curdist = 0; 
+		// stores the current distance if barn i was the opened barn
+		for(int pos = 0; pos < n; pos++) {
+			// the current barn we are processing is open_barn + pos
+			// however, since it is a circular array 
+			// the position may wrap around
+			// index = open_barn + pos
+			curdist += pos * cows[(open_barn + pos) % n];
 		}
-		sol = min(sol, cur);
+		ret = min(ret, curdist);
 	}
-	cout << sol << '\n';
+
+  cout << ret << '\n';
+
+  return 0;
 }
 
 ```
 
 </CPPSection>
+	
+<JavaSection>
+	
+## Implementation	
```

Comment:
move ## Implementation outside languagesection and remove ## Implementation from cppsection